google_guest_agent/cfg/cfg.go

```diff
@@ -0,0 +1,310 @@
+}
+
+// MetadataScripts contains the configurations of MetadataScripts section.
+type MetadataScripts struct {
+	DefaultShell      string `ini:"default_shell,omitempty"`
+	RunDir            string `ini:"run_dir,omitempty"`
+	Shutdown          bool   `ini:"shutdown,omitempty"`
+	ShutdownWindows   bool   `ini:"shutdown-windows,omitempty"`
+	Startup           bool   `ini:"startup,omitempty"`
+	StartupWindows    bool   `ini:"startup-windows,omitempty"`
+	SysprepSpecialize bool   `ini:"sysprep_specialize,omitempty"`
+}
+
+// NetworkInterfaces contains the configurations of NetworkInterfaces section.
+type NetworkInterfaces struct {
+	DHCPCommand  string `ini:"dhcp_command,omitempty"`
+	IPForwarding bool   `ini:"ip_forwarding,omitempty"`
+	Setup        bool   `ini:"setup,omitempty"`
+}
+
+// Snapshots contains the configurations of Snapshots section.
+type Snapshots struct {
+	Enabled             bool   `ini:"enabled,omitempty"`
+	SnapshotServiceIP   string `ini:"snapshot_service_ip,omitempty"`
+	SnapshotServicePort int    `ini:"snapshot_service_port,omitempty"`
+	TimeoutInSeconds    int    `ini:"timeout_in_seconds,omitempty"`
+}
+
+// Unstable contains the configurations of Unstable section. No long term stability or support
+// is guaranteed for configurations defined in the Unstable section. By default all flags defined
+// in this section is disabled and is intended to isolate under development features.
+type Unstable struct {
+	PAMLessAuthStack bool `ini:"pamless_auth_stack,omitempty"`
+}
+
+// WSFC contains the configurations of WSFC section.
+type WSFC struct {
+	Addresses string `ini:"addresses,omitempty"`
+	Enable    bool   `ini:"enable,omitempty"`
+	Port      string `ini:"port,omitempty"`
+}
+
+func defaultConfigFile(osName string) string {
+	if osName == "windows" {
+		return winConfigPath
+	}
+	return unixConfigPath
+}
+
+func defaultDataSources(extraDefaults []byte) []interface{} {
+	var res []interface{}
+	configFile := defaultConfigFile(runtime.GOOS)
+
+	if len(extraDefaults) > 0 {
+		res = append(res, extraDefaults)
+	}
+
+	return append(res, []interface{}{
+		[]byte(defaultConfig),
+		configFile,
+		configFile + ".distro",
+		configFile + ",template",
+	}...)
+}
+
+// Load loads default configuration and the configuration from default config files.
+func Load(extraDefaults []byte) error {
+	opts := ini.LoadOptions{
+		Loose:       true,
+		Insensitive: true,
+	}
+
+	sources := dataSources(extraDefaults)
+	cfg, err := ini.LoadSources(opts, sources[0], sources[1:]...)
+	if err != nil {
+		return fmt.Errorf("failed to load configuration: %+v", err)
+	}
+
+	sections := new(Sections)
+	if err := cfg.MapTo(sections); err != nil {
+		return fmt.Errorf("failed to map configuration to object: %+v", err)
+	}
+
+	instance = sections
+	return nil
+}
+
+// Get returns the configuration's instance previously loaded with Load().
+func Get() *Sections {
+	if instance == nil {
+		panic(fmt.Errorf("cfg package was not initialized, Load() " +
```

Comment:
as we are not using `recover()` and this would result into string representation only, we could use direct string message instead of `fmt.Errorf`

Reply:
Yep, we can. I'll get it updated. Thanks.

Reply:
Done. Thanks.